### DISCUSSION TITLE
1.2 编程要素

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#92] [Announcements]
# 1.2 编程要素

SICP Python 中文翻译

https://composingprograms.netlify.app/1/2

**Comment by jiangnan2005** [comment]
哇偶  我要第一个留名
一年之后我一定要成为编程大神

**Reply by ZC34** [reply]
加油

**Reply by washington66778899** [reply]
你好，可以一起学习吗？请求一起学习，不会拖后腿的

**Reply by Keta03** [reply]
可以一起么

**Reply by washington66778899** [reply]
可以，我加你，你给我你的联系方式

**Reply by Keta03** [reply]
Q487176493

**Reply by fshuaishuaishuai** [reply]
大神学到哪了

**Reply by 02-gxl** [reply]
成了吗孩子

**Comment by Archimondeeee** [comment]
我来解释里面的一些我不知道的专有名词啦
中缀表示法是一个通用的算术或逻辑公式表示方法， 操作符是以中缀形式处于操作数的中间（例：3 + 4），中缀表达式是人们常用的算术表示方法。
与前缀表达式（例：+ 3 4）或后缀表达式（例：3 4 +）相比，中缀表达式不容易被计算机解析，但仍被许多程序语言使用，因为它符合人们的普遍用法。
与前缀或后缀记法不同的是，中缀记法中括号是必需的。计算过程中必须用括号将操作符和对应的操作数括起来，用于指示运算的次序。

**Reply by happycen** [reply]
类似前序后序和中序吗

**Comment by visionary-5** [comment]
一起学习监督
q3314591463

**Comment by blue-RAT** [comment]
9/1：print()本身是一个输出值永远是none的函数，副作用是显示括号里本身的字符
一般来说会自动隐藏这个none的结果，但似乎只不显示最外边的？因为倒数第二层两个作为函数值出现的none被显示了。然后在print里可以用sep=’ ‘来改变间隔符号（默认是空格）
非纯函数：输入东西不只有返回值，会有副作用（ai说非纯函数可以有多个副作用但没太看懂）
以及python默认输出顺序是从下往上的

**Comment by sxc-323** [comment]
我也来一个，学习从今天开始

**Comment by Raychel-elf** [comment]
q3288365882，希望有很多志同道合的人一起学习，我刚开始学
